pkg/cluster/delete.go

```diff
@@ -63,3 +90,57 @@ func Delete(
 	}
 	return nil
 }
+
+func DeleteOverrideDir(appIf argoapp.ApplicationServiceClient, kubeClient *kubernetes.Clientset, argocdNs string, name string) error {
+	log := logpkg.GetLogger()
+	app, err := appIf.Get(context.Background(),
+		&apppkg.ApplicationQuery{
+			Name: &name,
+		})
+	if err != nil {
+		return fmt.Errorf("failed to get argocd application: %s", err)
+	}
+	typ := app.Labels["arlon-type"]
+	if typ == "cluster-app" {
```

Comment:
There's no need to check the type if we assume the caller has already done so. You can:
- Not check
- Still check, and return an error if it's not of the expected type.